code/game/gamemodes/clock_cult/clock_ratvar.dm

```diff
@@ -0,0 +1,240 @@
+			prey = null
+		else
+			if(prob(75))
+				dir_to_step_in = get_dir(src, prey)
+	for(var/i in 1 to 2)
+		loc = get_step(src, dir_to_step_in)
+
+/obj/structure/clockwork/massive/ratvar/narsie_act()
+	if(clashing)
+		return 0
+	clashing = TRUE
+	world << "<span class='heavy_brass'><font size=5>\"[pick("BLOOD GOD!!!", "NAR-SIE!!!", "AT LAST, YOUR TIME HAS COME!")]\"</font></span>"
+	world << "<span class='cult'><font size=5>\"<b>Ratvar?! How?!</b>\"</font></span>"
+	for(var/obj/singularity/narsie/N in range(7, src))
+		clash_of_the_titans(N) //IT'S TIME FOR THE BATTLE OF THE AGES
+		N.clashing = TRUE
+		break
+	return 1
+
+/obj/structure/clockwork/massive/ratvar/proc/clash_of_the_titans(obj/singularity/narsie/narsie) //IT'S TIME FOR A BATTLE OF THE ELDER GODS
+	var/winner = "Undeclared"
+	var/base_victory_chance = 0
+	while(TRUE)
+		world << 'sound/magic/clockwork/ratvar_attack.ogg'
+		sleep(5.2)
+		for(var/mob/M in mob_list)
+			if(M.client)
+				M.client.color = rgb(150, 100, 0)
+				spawn(1)
+					M.client.color = initial(M.client.color)
+			shake_camera(M, 4, 3)
+		var/r_success_modifier = (ticker.mode.servants_of_ratvar.len * 2) //2% for each cultist
+		var/n_success_modifier = (ticker.mode.cult.len * 2)
+		if(prob(base_victory_chance + r_success_modifier))
+			winner = "Ratvar"
+			break
+		sleep(rand(2,5))
+		world << 'sound/magic/clockwork/narsie_attack.ogg'
+		sleep(7.4)
+		for(var/mob/M in mob_list)
+			if(M.client)
+				M.client.color = rgb(200, 0, 0)
+				spawn(1)
+					M.client.color = initial(M.client.color)
+			shake_camera(M, 4, 3)
+		if(prob(base_victory_chance + n_success_modifier))
+			winner = "Nar-Sie"
+			break
+		base_victory_chance++ //The clash has a higher chance of resolving each time both gods attack one another
+	switch(winner)
+		if("Ratvar")
+			world << "<span class='heavy_brass'><font size=5>\"[pick("DIE! DIE! DIE!", "RAAAAAAAAAAAAAHH!", "FILTH!!!", "SUFFER!!!", "EBG SBE PRAGHEVRF NF V UNIR!!")]\"</font></span>"
+			world << "<span class='cult'><font size=5>\"<b>[pick("Nooooo...", "Not die. To y-", "Die. Ratv-", "Sas tyen re-")]\"</b></font></span>"
+			world << 'sound/magic/clockwork/anima_fragment_attack.ogg'
+			world << 'sound/magic/demon_dies.ogg'
+			clashing = FALSE
+			qdel(narsie)
+			return 1
+		if("Nar-Sie")
+			world << "<span class='cult'><font size=5>\"<b>[pick("Ha.", "Ra'sha fonn dest.", "You fool. To come here.")]</b>\"</font></span>" //Broken English
+			world << 'sound/magic/demon_attack1.ogg'
+			world << 'sound/magic/clockwork/anima_fragment_death.ogg'
+			narsie.clashing = FALSE
+			qdel(src)
+			return 1
+
+/atom/proc/ratvar_act() //Called on everything near Ratvar
+	return
+
+/turf/closed/wall/ratvar_act() //Walls and floors are changed to their clockwork variants
+	if(prob(20))
+		ChangeTurf(/turf/closed/wall/clockwork)
+/turf/open/floor/ratvar_act()
+	if(prob(20))
+		ChangeTurf(/turf/open/floor/clockwork)
+
+/obj/structure/window/ratvar_act() //Windows turn yellow
+	color = rgb(75, 53, 0)
```

Comment:
You may want a different yellow for shuttle windows, in the same way as narsie_act()